base/compiler/ssair/ir.jl

```diff
@@ -514,7 +515,7 @@ mutable struct IncrementalCompact
         new_new_nodes = NewNode[]
         pending_nodes = NewNode[]
         pending_perm = Int[]
-        return new(code, result, result_types, result_lines, result_flags, result_bbs, ssa_rename, bb_rename, used_ssas, late_fixup, perm, 1,
+        return new(code, result, result_types, result_lines, result_flags, result_bbs, ssa_rename, bb_rename, bb_rename, used_ssas, late_fixup, perm, 1,
             new_new_nodes, pending_nodes, pending_perm,
```

Comment:
I guess we set these to different arrays anyway in `cfg_simplify`, but maybe we should just initialize them as two separate arrays here? just so downstream code doesn't get tricked by this aliasing

Reply:
It intentionally uses the same array. The array is read only unless cfg transforms are enabled.